sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TableOutputResolver.scala

```diff
@@ -238,11 +238,17 @@ object TableOutputResolver {
 
     if (reordered.length == expectedCols.length) {
       if (matchedCols.size < inputCols.length) {
-        val extraCols = inputCols.filterNot(col => matchedCols.contains(col.name))
-          .map(col => s"${toSQLId(col.name)}").mkString(", ")
-        throw QueryCompilationErrors.incompatibleDataToTableExtraStructFieldsError(
-          tableName, colPath.quoted, extraCols
-        )
+        if (colPath.isEmpty) {
+          val cannotFindCol = expectedCols.filter(col => !matchedCols.contains(col.name)).head.name
+          throw QueryCompilationErrors.incompatibleDataToTableCannotFindDataError(tableName,
```

Comment:
when we reach here, why it means missing col for top-level columns, but extra column for nested fields?

Reply:
where do we check missing cols error for nested fields?

Reply:
The code here is so convoluted...

`if (reordered.length == expectedCols.length)` I think this means there is no missing col? For every table column, we can find a match in the input columns.

Reply:
>  eg: table [x, y, z], inputcol: [x, y, k],

If `z` has default value, I think it means extra column error?

Reply:
missing column error should only be thrown if that missing column does not have a default value.